### PR TITLE
feat: raise if both `closed` and `by` are passed to `rolling_*` aggregations

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes https://github.com/pola-rs/polars/issues/15097

### CODE REVIEW

File: crates/polars-time/src/chunkedarray/rolling_window/mod.rs [hunk 1]
```diff
@@ -98,25 +105,7 @@ impl From<RollingOptions> for RollingOptionsImpl<'static> {
             tz: None,
             closed_window: None,
             fn_params: options.fn_params,
-        }
-    }
-}
-
-impl From<RollingOptions> for RollingOptionsFixedWindow {
```
Comment (MarcoGorelli) [inline crates/polars-time/src/chunkedarray/rolling_window/mod.rs:105]: this is unused - OK to just remove?

File: crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs [hunk 2]
```diff
@@ -48,8 +48,7 @@ where
     let arr = ca.downcast_iter().next().unwrap();
     // "5i" is a window size of 5, e.g. fixed
     let arr = if options.window_size.parsed_int {
-        let options: RollingOptionsFixedWindow = options.into();
-        check_input(options.window_size, options.min_periods)?;
```
Comment (MarcoGorelli) [inline crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs:52]: I'm moving the `check_input` into `try_from`, so the `try_into` covers this (and other places where this is used won't need to remember to call `check_input`)

Comment (ritchie46) [reply crates/polars-time/src/chunkedarray/rolling_window/dispatch.rs:52]: Yes, good one.

File: crates/polars-time/src/chunkedarray/rolling_window/mod.rs (hunk 1 above)
Comment (ritchie46) [reply crates/polars-time/src/chunkedarray/rolling_window/mod.rs:105]: Yes, there is a no unused code policy. :police_officer:

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**